lib/routes/oauth.js

```diff
@@ -76,6 +78,49 @@ module.exports = (log, config, oauthdb) => {
         return oauthdb.createAuthorizationCode(sessionToken, request.payload)
       }
     },
+    {
+      method: 'POST',
+      path: '/oauth/token',
+      options: {
+        auth: {
+          // XXX TODO: This route must also be able to accept 'client_secret' as Basic Auth in header.
+          mode: 'optional',
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: Joi.alternatives.try(
+            oauthdb.api.grantTokensFromAuthorizationCode.opts.validate.payload,
+            oauthdb.api.grantTokensFromRefreshToken.validate.payload,
+            oauthdb.api.grantTokensFromAssertion.opts.validate.payload.keys({
+              assertion: Joi.forbidden()
+            })
+          )
+        },
+        response: {
+          schema: Joi.alternatives.try(
+            oauthdb.api.grantTokensFromAuthorizationCode.opts.validate.response,
+            oauthdb.api.grantTokensFromRefreshToken.validate.response,
+            oauthdb.api.grantTokensFromAssertion.opts.validate.response
+          )
+        }
+      },
+      handler: async function (request) {
+        const sessionToken = request.auth.credentials
```

Comment:
should we use `const credentials` here? instead of `sessionToken?

Reply:
I didn't, because it *must* be a sessionToken; if you're using a refreshToken is gets passed in the request body.